### DISCUSSION TITLE
The initial value of some hooks are confusing

### DISCUSSION THREAD

**Discussion by colgin** [react-hookz/web#1307] [Q&A]
there are some hooks like useCookieValue, useStorageValue ... , the initial value of useState is related to useFirstMountState, see [react-hookz/web search useFirstMountState](https://github.com/search?q=repo%3Areact-hookz%2Fweb%20useFirstMountState&type=code).

eg.
```js
function useCookieValue() {
  // ...
  const isFirstMount = useFirstMountState();
  const [state, setState] = useState<string | null | undefined>(
    isFirstMount && initializeWithValue ? methods.current.fetchVal() : undefined
  );
  // ...
}
```

useFirstMountState is pretty easy
```js
export function useFirstMountState(): boolean {
	const isFirstMount = useRef(true);

	useEffect(() => {
		isFirstMount.current = false;
	}, []);

	return isFirstMount.current;
}
```
It means the first time useState executed with isFirstMount = true, after first render, isFirstMount is always false.

so, what's the difference with this
```js
const [state, setState] = useState<string | null | undefined>(
  true && initializeWithValue ? methods.current.fetchVal() : undefined
 );
```

**Comment by xobotyi** [comment]
Noone will be changing the `initializeWithValue` option after first render, therefore value fetching will be performed _each_ render, for cookie it is pretty straightforward, but for storages it envolves deserialization
